src/main/java/io/proximax/sdk/model/transaction/builder/LockFundsTransactionBuilder.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 ProximaX Limited. All rights reserved.
+ * Use of this source code is governed by the Apache 2.0
+ * license that can be found in the LICENSE file.
+ */
+package io.proximax.sdk.model.transaction.builder;
+
+import java.math.BigInteger;
+
+import io.proximax.sdk.model.mosaic.Mosaic;
+import io.proximax.sdk.model.transaction.AccountLinkTransaction;
+import io.proximax.sdk.model.transaction.LockFundsTransaction;
+import io.proximax.sdk.model.transaction.SignedTransaction;
+import io.proximax.sdk.model.transaction.TransactionType;
+import io.proximax.sdk.model.transaction.TransactionVersion;
+
+/**
+ * builder for {@link LockFundsTransaction}
+ */
+public class LockFundsTransactionBuilder
+      extends TransactionBuilder<LockFundsTransactionBuilder, LockFundsTransaction> {
+
+   private Mosaic mosaic;
+   private BigInteger duration;
+   private SignedTransaction signedTransaction;
+
+   public LockFundsTransactionBuilder() {
+      super(TransactionType.LOCK, TransactionVersion.LOCK.getValue());
+   }
+
+   @Override
+   protected LockFundsTransactionBuilder self() {
+      return this;
+   }
+
+   @Override
+   public LockFundsTransaction build() {
+      // use or calculate maxFee
+      BigInteger maxFee = getMaxFee().orElseGet(
+            () -> getMaxFeeCalculation(AccountLinkTransaction.calculatePayloadSize()));
```

Comment:
copy/paste?

Reply:
good find! will fix that. funny thatbtests passed...
ah man you can not imagine the amount of copy pasting I am doing there 🙂

Reply:
fixed